src/Microsoft.ML.Transforms/Text/NgramHashingTransformer.cs

```diff
@@ -1028,12 +970,193 @@ public VBuffer<ReadOnlyMemory<char>>[] SlotNamesMetadata(out ColumnType[] types)
                         types[iinfo] = new VectorType(TextType.Instance, vec.Length);
                     }
                 }
                 return values;
             }
         }
     }
+
+    /// <summary>
+    /// Produces a bag of counts of ngrams (sequences of consecutive words of length 1-n) in a given text.
+    /// It does so by hashing each ngram and using the hash value as the index in the bag.
+    ///
+    /// <see cref="NgramHashingEstimator"/> is different from <see cref="WordHashBagEstimator"/> in a way that <see cref="NgramHashingEstimator"/>
+    /// takes tokenized text as input while <see cref="WordHashBagEstimator"/> tokenizes text internally.
+    /// </summary>
+    public sealed class NgramHashingEstimator : IEstimator<NgramHashingTransformer>
+    {
+        internal static class Defaults
+        {
+            internal const int NgramLength = 2;
+            internal const bool AllLengths = true;
+            internal const int SkipLength = 0;
+            internal const int HashBits = 16;
+            internal const uint Seed = 314489979;
+            internal const bool RehashUnigrams = false;
+            internal const bool Ordered = true;
+            internal const int InvertHash = 0;
+        }
+
+        private readonly IHost _host;
+        private readonly NgramHashingTransformer.ColumnInfo[] _columns;
+
+        /// <summary>
+        /// Produces a bag of counts of hashed ngrams in <paramref name="inputColumn"/>
+        /// and outputs ngram vector as <paramref name="outputColumn"/>
+        ///
+        /// <see cref="NgramHashingEstimator"/> is different from <see cref="WordHashBagEstimator"/> in a way that <see cref="NgramHashingEstimator"/>
+        /// takes tokenized text as input while <see cref="WordHashBagEstimator"/> tokenizes text internally.
+        /// </summary>
+        /// <param name="env">The environment.</param>
+        /// <param name="inputColumn">The column containing text to compute bag of word vector.</param>
+        /// <param name="outputColumn">The column containing bag of word vector. Null means <paramref name="inputColumn"/> is replaced.</param>
+        /// <param name="hashBits">Number of bits to hash into. Must be between 1 and 30, inclusive.</param>
+        /// <param name="ngramLength">Ngram length.</param>
+        /// <param name="skipLength">Maximum number of tokens to skip when constructing an ngram.</param>
+        /// <param name="allLengths">Whether to include all ngram lengths up to <paramref name="ngramLength"/> or only <paramref name="ngramLength"/>.</param>
+        /// <param name="seed">Hashing seed.</param>
+        /// <param name="ordered">Whether the position of each source column should be included in the hash (when there are multiple source columns).</param>
+        /// <param name="invertHash">Limit the number of keys used to generate the slot name to this many. 0 means no invert hashing, -1 means no limit.</param>
```

Comment:
>Limit the number of keys used to generate the slot name to this many. [](start = 37, length = 69)

I am not certain that I understand the relationship between number of keys used to generate the slot name and invert hashing, would be helpful to have some more details.